src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Queries/GetProvider/GetProviderQueryHandler.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.CommitmentsV2.Data;
+
+namespace SFA.DAS.CommitmentsV2.Application.Queries.GetProvider
+{
+    public class GetProviderQueryHandler : IRequestHandler<GetProviderQuery, GetProviderQueryResult>
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _db;
+
+        public GetProviderQueryHandler(Lazy<ProviderCommitmentsDbContext> db)
+        {
+            _db = db;
+        }
+
+        public async Task<GetProviderQueryResult> Handle(GetProviderQuery request, CancellationToken cancellationToken)
+        {
+            var result = await _db.Value.Providers
+                .Where(p => p.UkPrn == request.ProviderId)
+                .Select(p => new GetProviderQueryResult(p.UkPrn, p.Name))
+                .SingleOrDefaultAsync(cancellationToken);
+            
```

Comment:
You could refactor this instead to:

```
var dbResult = await _db.Value.Providers
       .FindAsync(new object []{request.ProviderId}, cancellationToken);
            
       return new GetProviderQueryResult(dbResult.UkPrn, dbResult.Name);
```

Reply:
Refactoring to this causes a null reference exception at dbResult.UkPrn and dbResult.Name if the db doesn't have an instance of that provider. This can be fixed by some conditional checking but it makes it less clean. Thoughts?

Reply:
Both approaches are valid but for consistency purposes, we'll leave it as originally written.

Reply:
Leaving as per per discussion with Dan, Chris and Ian.